addon/-private/system/model/internal-model.js

```diff
@@ -343,7 +343,7 @@ export default class InternalModel {
       // instances with the injections applied
       let createOptions = {
         store: this.store,
-        _internalModel: this,
+        __setupInternalModel: this,
```

Comment:
Can you add a comment here also? So that it's clear that this `__setupInternalModel` property must exist before any user specified properties. In particular, we care the most about user specified properties that are relationships.